lms/djangoapps/api_manager/courses/tests.py

```diff
@@ -2148,6 +2148,19 @@ def test_courses_data_metrics(self):
         response = self.do_get(course_metrics_uri)
         self.assertEqual(response.status_code, 404)
 
+    def test_course_data_metrics_user_group_filter(self):
```

Comment:
@msaqib52 I think we should create separate tests for each scenario we are covering instead of putting everything in a single test.